generate_diagrams.py

```
@@ -0,0 +1,200 @@
+        generate_import_graph()
+    except RuntimeError as e:
+        print(f"Error generating import graph: {e}")
+
+    print("\nDiagram generation complete!")
+    print("\nTo view the diagrams:")
+    print("1. Open the Mermaid diagrams in ARCHITECTURE.md (renders on GitHub)")
+    print("2. Check docs/diagrams/ for generated SVG files")
+
+
+if __name__ == "__main__":
+    main()
```

Pylint (3.3.7)

[convention] Final newline missing

(C0304)